casper/src/main/scala/io/casperlabs/casper/api/BlockAPI.scala

```
@@ -289,14 +289,11 @@ object BlockAPI {
       ) => F[A]
   ): F[A] =
     for {
-      dag         <- MultiParentCasper[F].blockDag
-      header      = block.header.getOrElse(Header.defaultInstance)
-      version     = header.version
-      deployCount = header.deployCount
-      tsHash = ProtoUtil.tuplespace(block) match {
-        case Some(hash) => hash
-        case None       => ByteString.EMPTY
-      }
+      dag                      <- MultiParentCasper[F].blockDag
+      header                   = block.header.getOrElse(Header.defaultInstance)
+      version                  = header.version
+      deployCount              = header.deployCount
+      postStateHash            = ProtoUtil.tuplespace(block).getOrElse(ByteString.EMPTY)
```

There's a `ProtoUtil.preStateHash(block)` method. Let's use it. It's better to throw an exception about missing field in this case than pretend it's `ByteString.EMPTY`.